### ISSUE TITLE
Picture taken with this plugin and then saved to the file system looks blurry 

### ISSUE THREAD

**Issue by blue2609** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#415]
Hi guys, hope you're all doing well. 

I just realised this issue only yesterday as I was trying to finalise a small camera app project that I have. Before I go into details, I have to let you know that I'm testing my app on my **_Google Pixel 1st Gen_** which is running **_Android 8.1.0_** and has **_12.3MP_** back camera. 

My app uses this function below to take each picture:

```
  CameraPreview.takePicture({quality: 100}, function(picture){ 
       //convert the base 64 encoded string to BLOB 
       //save the newly created BLOB to the android device's file system storage
  }
```

Just like how I described it above, I convert the **_base 64 encoded string_** that was given as parameter to the success callback function into **BLOB**. This **BLOB** is then written into a file system storage inside an android device. 

The function which was used to convert the 64 bit encoded string to BLOB is this:

```
function b64toBlob(b64Data, contentType, sliceSize) {
      contentType = contentType || '';
      sliceSize = sliceSize || 512;

      var byteCharacters = atob(b64Data);
      var byteArrays = [];

      for (var offset = 0; offset < byteCharacters.length; offset += sliceSize) {
          var slice = byteCharacters.slice(offset, offset + sliceSize);

          var byteNumbers = new Array(slice.length);
          for (var i = 0; i < slice.length; i++) {
              byteNumbers[i] = slice.charCodeAt(i);
          }

          var byteArray = new Uint8Array(byteNumbers);

          byteArrays.push(byteArray);
      }

    var blob = new Blob(byteArrays, {type: contentType});
    return blob;
}
```

Problem is, while this method is working, the picture saved inside the file system storage is blurry and it looks low quality (_it's only **2.1 MP** and its resolution is **1920 X 1080**_). I don't understand why this happens as my camera is capable of capturing a very sharp **_12.3MP_**, which translates to a picture resolution of **_3036 X 4048_**. 

This is very far from ideal and as I was just playing around with the camera plugin for a week, I haven't found any solution to this. 

Does anyone out there know how to fix this issue?

**Comment by ngt14** [comment]
hi @blue2609, i have this same issue but i have a solution.
i think this plugin by default take 1st resolution of the camera (very bad resolution) but you can use function **getSupportedPictureSizes(cb, [errorCallback])** and select resolution according to your needs by adding width and height parameters on function **CameraPreview.takePicture()**

**Comment by blue2609** [comment]
@ngt14 

Yeah I tried it already mate, it doesn't work. Well, okay, maybe it makes the picture taken a teeny bit sharper but it's still blurry. 

I tried comparing the image taken with this plugin VS image taken with the stock google camera that I have on my google Pixel and it's just so much sharper with the google camera. 

Any idea why or if it's possible to fix it? The only other alternative I can see right now is to start from scratch and use Android Studio to code a new camera app.

**Comment by nandanshan** [comment]
@blue2609 
Have you tried passing desired width and height?

```
private pictureOpts: CameraPreviewPictureOptions = {
        width: 3120,
        height: 4160,
        quality: 100
      };
```

the plugin will consider the resolution which is nearest to the width and height which you had passed

**Comment by Choppel** [comment]
If you omit width and height, the plugin does not use the maximum dimensions available.

In www/CameraPreview.js at lines 65 to 66 it uses 0 as width and height, if they are omitted. Inside src/android/CameraActivity.java at lines 448 to 454 it says that, if width and height are 0, the function returns the size from supportedSizes that is closest to 2MP.

Just use the solution @nandanshan suggested and you should be fine.

**Comment by PabbleDabble** [comment]
@cryptoshan  Have you noticed that when you see the height and width like this that the image file size becomes very large, or at least much larger than the native camera images taken with out the preview?
